File: b3323rec/app/src/main/java/com/example/b3323rec/MainActivity.java
package com.example.b3323rec;
//OnClickListenerLittle
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    ArrayList<LittleItem> listLittleItems;
    RecyclerView recyclerView;
    LittleAdapter littleAdapter;
    RecyclerView.LayoutManager manager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = findViewById(R.id.recycleView);

        listLittleItems = new ArrayList<>();
        listLittleItems.add(new LittleItem("little 1"));
        listLittleItems.add(new LittleItem("little 2"));
        listLittleItems.add(new LittleItem("little 3"));
        listLittleItems.add(new LittleItem("little 4"));

        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        littleAdapter = new LittleAdapter(listLittleItems);
        recyclerView.setAdapter(littleAdapter);

        littleAdapter.setOnClickListener(new LittleAdapter.OnClickListenerLittle() {
            @Override
            public void onClick(int position) {
                Toast.makeText(MainActivity.this, "Position: " + position, Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: l2152alert/app/src/main/java/com/example/freon/l2152alert/MainActivity.kt
package com.example.freon.l2152alert

import android.content.DialogInterface
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AlertDialog

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mMultiDialog()
    }

    private fun mSimpleDialog(){
        val builder = AlertDialog.Builder(this)
        builder.setTitle("Week")
        builder.setMessage("day")
        builder.setNeutralButton("info"){ dialogInterface: DialogInterface,
                                          i: Int ->
            Toast.makeText(this,"Info", Toast.LENGTH_LONG).show()
        }
        builder.show()
    }

    private fun mMultiDialog(){
        val builder = AlertDialog.Builder(this)
        builder.setTitle("Week")
        builder.setMultiChoiceItems(R.array.string_array,null){
            dialogInterface: DialogInterface,
            i: Int,
            b: Boolean ->
            Toast.makeText(this,
                    " "+ i + ":" + b, Toast.LENGTH_LONG).show()

        }
        builder.setNeutralButton("info"){ dialogInterface: DialogInterface,
                                          i: Int ->
        Toast.makeText(this,"Info", Toast.LENGTH_LONG).show()
        }
        builder.show()
    }
}